v2/internal/controllers/crd_eventgrid_domain_test.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	eventgrid "github.com/Azure/azure-service-operator/v2/api/eventgrid/v1alpha1api20200601"
+	storage "github.com/Azure/azure-service-operator/v2/api/storage/v1alpha1api20210401"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+func Test_EventGrid_Domain(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	publicNetworkAccess := eventgrid.DomainPropertiesPublicNetworkAccessEnabled
+
+	// Create a domain
+	domain := &eventgrid.Domain{
+		ObjectMeta: tc.MakeObjectMeta("domain"),
+		Spec: eventgrid.Domains_Spec{
+			Location:            tc.AzureRegion,
+			Owner:               testcommon.AsOwner(rg),
+			PublicNetworkAccess: &publicNetworkAccess,
+		},
+	}
+
+	// Create a storage account to use as destination
+	accessTier := storage.StorageAccountPropertiesCreateParametersAccessTierHot
+	namer := tc.Namer.WithSeparator("") // storage account rules are different
+	acctName := namer.GenerateName("dest")
+	acct := &storage.StorageAccount{
+		ObjectMeta: tc.MakeObjectMetaWithName(acctName),
+		Spec: storage.StorageAccounts_Spec{
+			Location: tc.AzureRegion,
+			Owner:    testcommon.AsOwner(rg),
+			Kind:     storage.StorageAccountsSpecKindStorageV2,
+			Sku:      storage.Sku{Name: storage.SkuNameStandardLRS},
+			// TODO: They mark this property as optional but actually it is required
+			AccessTier: &accessTier,
+		},
+	}
+
+	// TODO: Getting this is SUPER awkward
```

Comment:
You could try this with the k8s-level reference - it should be something like `acctReference := tc.MakeReferenceFromResource(acct)`. It should run with the same recording, right?

Reply:
Will try it and see…

Reply:
it worked! 😁